### PR TITLE
Paranet support

### PR DESCRIPTION
**Pull request by NZT48**

### CODE REVIEW

File: managers/paranet-operations-manager.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+        } = this.inputService.getParanetDeployIncentivesContractArguments(options);
+
+        this.validationService.validateDeployIncentivesContract(
+            paranetUAL,
+            blockchain,
+            tracToNeuroEmissionMultiplier,
+            operatorRewardPercentage,
+            incentivizationProposalVotersRewardPercentage,
+        );
+        if(incentiveType === 'Neuroweb') {
```
Comment (Mihajlo-Pavlovic) [inline managers/paranet-operations-manager.js:57]: Can we crate some constanst in constasts which will emulate enum like INCENTIVE_TYPE.NEUROWEB = 1and INCENTIVE_TYPE.NOT_NEUROWEB = 0

File: managers/paranet-operations-manager.js [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+                    incentivizationProposalVotersRewardPercentage,
+                },
+                blockchain
+            );
+
+            const paranetId = ethers.keccak256(
+                ethers.solidityPacked(['address', 'uint256'], [contract, tokenId]),
+            );
+
+            const neuroIncentivesPoolAddress = await this.blockchainService.getNeuroIncentivesPoolAddress(paranetId, blockchain);
```
Comment (Mihajlo-Pavlovic) [inline managers/paranet-operations-manager.js:74]: We should check if we got address not null or 0x0 not to burn coins if error happens

File: services/blockchain-service/blockchain-service-base.js [hunk 3]
```diff
@@ -216,7 +225,9 @@ class BlockchainServiceBase {
         return this[blockchain.name].contracts[blockchain.hubContract][contractName];
     }
 
-    async createAsset(requestData, blockchain, stepHooks = emptyHooks) {
+    // Knowledge assets operations
+
+    async createAsset(requestData, paranetKaContract, paranetTokenId, blockchain, stepHooks = emptyHooks) {
```
Comment (Mihajlo-Pavlovic) [inline services/blockchain-service/blockchain-service-base.js:230]: We changed signature here, did we update it everywhere it is used?

Comment (Mihajlo-Pavlovic) [reply services/blockchain-service/blockchain-service-base.js:230]: Maybe rewrite it to be 
createAsset(requestData, blockchain, stepHooks = emptyHooks, paranetKaContract = null, paranetTokenId = null)

File: services/blockchain-service/blockchain-service-base.js [hunk 4]
```diff
@@ -600,6 +862,18 @@ class BlockchainServiceBase {
         return web3.eth.getBlock(blockNumber);
     }
 
+    // TODO: Temporary to remove
+    async sendTokens( targetAddress , blockchain) {
```
Comment (Mihajlo-Pavlovic) [inline services/blockchain-service/blockchain-service-base.js:866]: Why is here this todo?

Comment (NZT48) [reply services/blockchain-service/blockchain-service-base.js:866]: So I don't forget to remove it xD

File: managers/asset-operations-manager.js [hunk 5]
```diff
@@ -1349,6 +1374,41 @@ class AssetOperationsManager {
...
+
+    /**
+     * Add knowledge asset to a paranet.
+     * @async
+     * @param {string} UAL - The Universal Asset Locator of the knowledge asset.
+     * @param {string} paranetUAL - The Universal Asset Locator of the Paranet.
+     * @param {Object} [options={}] - Additional options for adding tokens.
+     * @returns {Object} An object containing the UAL and operation status.
+     */
+    async submitToParanet(UAL, paranetUAL, options = {}) {
```
[the remark this answers is not in the document]
Comment (NZT48) [reply managers/asset-operations-manager.js:1386]: The logic was there were 2 ways of adding KA to the Paranet, one is through regular dkg.asset.create with paranet argument, and as this function is in asset manager, we said lets have other function that adds KAs to paranet to be in the asset manager. Also it sounded more descriptive that dkg.asset.submitToParanet(UAL) means adding asset to the paranet.

File: managers/paranet-operations-manager.js (hunk 2 above)
Comment (NZT48) [reply managers/paranet-operations-manager.js:105]: That sending tokens were testing part, it is removed

File: services/blockchain-service/blockchain-service-base.js (hunk 3 above)
Comment (NZT48) [reply services/blockchain-service/blockchain-service-base.js:230]: It is updated everywhere, we use it only internally on 2 places

File: managers/paranet-operations-manager.js (hunk 1 above)
Comment (NZT48) [reply managers/paranet-operations-manager.js:88]: Why would we assign numbers and have not neuroweb?

Comment (Mihajlo-Pavlovic) [reply managers/paranet-operations-manager.js:88]: Thats also good, just change it to be constant

File: services/blockchain-service/blockchain-service-base.js (hunk 4 above)
Comment (Mihajlo-Pavlovic) [reply services/blockchain-service/blockchain-service-base.js]: Thats what PR review are for :D

**Review by Mihajlo-Pavlovic — APPROVED** [review]

**Pull request merged**